src/search/dtos/search-all-mentors.dto.ts

```diff
@@ -0,0 +1,60 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { User } from 'src/users/entities/user.entity';
+import { UserIntro } from 'src/users/entities/user-intro.entity';
+import { UserImage } from 'src/users/entities/user-image.entity';
+
+export class SearchAllMentorsDto
+  implements Pick<User, 'id' | 'name' | 'userImage' | 'userIntro' | 'isMentor'>
+{
+  @ApiProperty({
+    description: '멘토 유저 고유 id',
+  })
+  id: number;
+
+  @ApiProperty({
+    description: '멘토 유저 이름',
+  })
+  name: string;
+
+  @ApiProperty({
+    description: '멘토 유저 이미지 object',
+    type: 'object',
+    properties: {
+      imageUrl: {
+        description: '멘토 유저 이미지 url',
+        type: 'string',
+      },
+    },
+  })
+  userImage: UserImage;
+
+  @ApiProperty({
+    description: '멘토 유저 소개 object',
+    type: 'object',
+    properties: {
+      mainField: {
+        description: '대표 멘토링 분야',
+        type: 'string',
+      },
+      introduce: {
+        description: '멘토 소개',
+        type: 'string',
+      },
+    },
+  })
+  userIntro: UserIntro;
```

Comment:
```suggestion
  @ApiProperty({
    description: '멘토 유저 소개 object',
    type: UserIntro
  userIntro: UserIntro;
```
userIntro 가 class 로 존재한다면 다음처럼 가능할거같아요.

Reply:
앗 userIntro에서 mainField와 introduce만을 select 해서 두개의 프로퍼티만을 넣었습니다! 타입도 Pick으로 수정해줬습니다.